src/devices/Camera/Camera/ProcessRunner.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Iot.Device.Camera.Settings;
+
+namespace Iot.Device.Camera;
+
+/// <summary>
+/// Controls an external process
+/// </summary>
+public class ProcessRunner : IDisposable
+{
+    private readonly ProcessSettings _processSettings;
+    private CancellationTokenSource _cts;
+    private Process? _process = null;
+
+    /// <summary>
+    /// Creates an instance of the ProcessRunner by getting the
+    /// Process Settings
+    /// </summary>
+    public ProcessRunner(ProcessSettings processSettings)
+    {
+        _cts = new CancellationTokenSource();
+        _processSettings = processSettings;
+    }
+
+    /// <summary>
+    /// The PID of the process that is running or null
+    /// </summary>
+    public int? PId => _process?.Id;
+
+    /// <summary>
+    /// Dispose the active running process
+    /// </summary>
+    public void Dispose()
+    {
+        _cts.Cancel();
+        Kill();
+    }
+
+    /// <summary>
+    /// Kill the process
+    /// </summary>
+    public void Kill()
+    {
+        try
+        {
+            if (_process != null && !_process.HasExited)
+            {
+#if NETSTANDARD2_0
+                _process.Kill();
+#else
+                _process.Kill(true);
+#endif
+                if (!_process.HasExited)
+                {
+                    _process.WaitForExit(5000);
+                }
+
+                _process.Dispose();
+            }
+        }
+        catch (Exception)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Retrieves the full command line equivalent to the one run from this class.
+    /// The working directory is either ProcessSettings.WorkingDirectory or, if null,
+    /// the Environment.CurrentDirectory.
+    /// The working directory is important when referring to files from the command line.
+    /// </summary>
+    public string GetFullCommandLine(string[] arguments)
```

Comment:
I have found that we haven't enabled the "Return value must be documented" stylecop rule. I'm going to enable it, but I need to fix quite a bunch of comments to do so.

Reply:
I hope I didn't forget any of them, but it was an easy but long work

Reply:
When using Visual Studio, the params, the return and whatever else including exceptions are added automatically, so it makes life easier! Also, with copilot, most of the comments can be done automatically with looking at the result.
When developing, just disable the rules for comments and activate it back once you're done and create the intellisense comments at once.

Reply:
Please note that stylecop does not warn about the missing comment for parameters in addition to the return type.

> When using Visual Studio, the params, the return and whatever else including exceptions are added automatically

I know, but every company has different ruleset and when I write code sometimes I jump from one style to another. Since VS did not complain, I committed without the other set of comments.